packages/perseus-editor/src/util/modernize-widgets-utils.ts

```diff
@@ -0,0 +1,205 @@
+                        ? widgets[key].options.maxError
+                        : 0,
+                    status: "correct", // Input-number only allows correct answers
+                    message: "",
+                },
+            ];
+
+            // Add the required answerForms if provided/applicable
+            if (provideAnswerForm) {
+                answers[0].answerForms = [mathFormat];
+            }
+
+            // Update the options prop to match the numeric-input widget format
+            const upgradedWidgetOptions = {
+                answers,
+                size: widgets[key].options.size,
+                coefficient: false, // input-number doesn't have a coefficient prop
+                labelText: "", // input-number doesn't have a labelText prop
+                static: false, // static is always false for numeric-input
+                rightAlign: widgets[key].options.rightAlign || false,
+            };
+            const upgradedWidget: NumericInputWidget = {
+                options: upgradedWidgetOptions,
+                type: "numeric-input",
+            };
+            const newWidgetName = renameMap[key];
+            // Create the new key entry
+            widgets[newWidgetName] = upgradedWidget;
+
+            // We need to delete the old widget key
+            delete widgets[key];
+        }
+    }
+    return widgets;
+};
+
+// Create a map of the old input-number keys to the new numeric-input keys
+const getInputNumberRenameMap = (json: PerseusRenderer): WidgetRenameMap => {
+    const numericRegex = /(?<=\[\[\u2603 )(numeric-input \d+)(?=\]\])/g;
+    const inputNumberRegex = /(?<=\[\[\u2603 )(input-number \d+)(?=\]\])/g;
+
+    // Get all the content strings within the json, which might be nested within widgets
+    const allContentStrings = json.content;
+
+    // Loop through the content strings to get all the input-number widgets
+    const renameMap: WidgetRenameMap = {};
+    const inputNumberMatches: string[] = [
+        ...(allContentStrings.match(inputNumberRegex) || []),
+    ];
+
+    // We want to count any pre-existing numeric-input widgets
+    // so that we can start the new ids at the next number
+    const numericMatches: string[] = [
+        ...(allContentStrings.match(numericRegex) || []),
+    ];
+    let currentNumericCount = 1;
+    if (numericMatches.length > 0) {
+        for (const match of numericMatches) {
+            const id = parseInt(match.split(" ")[1], 10);
+            if (id >= currentNumericCount) {
+                currentNumericCount = id + 1; // We want to start at the next highest number
+            }
+        }
+    }
+
+    // Now that we have all the necessary information, we can create the renameMap
+    for (const match of inputNumberMatches) {
+        const oldKey = match;
+        const newKey = `numeric-input ${currentNumericCount}`;
+        renameMap[oldKey] = newKey;
+        currentNumericCount++;
+    }
```

Comment:
Nit: Definitely a style thing...
```
inputNumberMatches.forEach((match, index) => {
    renameMap[match] = `numeric-input ${currentNumericCount + index}`;
});
```

I also appreciate the explicit-ness of the variable names and the clear steps, so definitely up to which appeals to you more.